### ISSUE TITLE
Exception EFBIG

### ISSUE THREAD

**Issue by hakankaraduman** [hgouveia/node-downloader-helper#2]
Thanks for this package. It works really well.

I try to download a file of size 2.67 GB. It gives error EFBIF: File too larget at around 84.5%. Is there a file size limitation and can I change it?

My node app runs on raspberry bi/raspbian jesse.

**Comment by hgouveia** [comment]
@hakankaraduman Thanks you very much!

The library doesn't has any file size limitation, also doesn't store the data in memory since it write the data directly to the file as soon it receives, but i will need to double check that, what i can think of  :

1. the storage is full
2. the storage is FAT32 partition format
  
Are you able to share the link of the file is causing you troubles so i can debug better?

**Comment by hakankaraduman** [comment]
Yes the sd card is fat32 formatted but the file size is not over 4 gb. 
Link is just a video I downloaded from youtube and uploaded to my server, just to test downloading a big file.
Here is the link to it: https://hwnodestorage.blob.core.windows.net/releases/Beautiful HD Aquarium Video - Georgia Aquarium (Ocean Voyager I)-zJ7hUvU-d2Q.mp4

**Comment by hgouveia** [comment]
@hakankaraduman i was testing your file,  i was able to download it from a FAT32 storage without any issue, i also tested this on a raspberry pi with not issues either, additionally i downloaded a 7GB , 10gb file , not issues there

It seems the issue is not related with the Downloader itself, i will suggest to try in another location or verify your drive for errors , sometimes drives with errors could notify false sizes to the OS
